lib/isafw/isaplugins/ISA_la_plugin.py

```diff
@@ -144,11 +170,26 @@ def write_report_xml(self):
 
     def write_report_unwanted(self):
         if os.path.isfile (self.report_name + "_unwanted"):
+            with open(self.logfile, 'a') as flog:
+                flog.write("image_pkgs: " + str(self.image_pkgs) + "\n")
+                flog.write("self.la_plugin_image_whitelist: " + str(self.la_plugin_image_whitelist) + "\n")
+                flog.write("self.la_plugin_image_blacklist: " + str(self.la_plugin_image_blacklist) + "\n")
             with open(self.report_name, 'a') as fout:
                 with open(self.report_name + "_unwanted", 'r') as f:
                     fout.write("\n\nPackages that violate mandatory license requirements:\n")
                     for line in f:
-                        fout.write(line)
+                        line = line.strip()
+                        pkg_name = line.split(':',1)[0]
+                        if (not self.image_pkgs):
+                            fout.write(line + " from image name not avaliable \n")
+                            continue
+                        for pkg_info in self.image_pkgs:
+                            if (pkg_info.split()[0] == pkg_name): 
+                                if self.la_plugin_image_whitelist and (pkg_info.split()[2] not in self.la_plugin_image_whitelist):
+                                    continue
+                                if self.la_plugin_image_blacklist and (pkg_info.split()[2] in self.la_plugin_image_blacklist):
+                                    continue
```

Comment:
I just tried to follow the same logic on whitelist/blacklist variable that I already have for plugins. Otherwise variables would work differently and it might be even more confusing. Before whitelist meant a whitelist of plugins you want to run and blacklist meant that "do not run" certain plugins, so I followed the same logic here.